src/services/openAiTranslate.ts

```diff
@@ -0,0 +1,54 @@
+import { OpenAiEngine } from "../types/settings";
+import { TranslateFn } from "./translate";
+
+export const translate: TranslateFn<OpenAiEngine> = async (
+  text: string,
+  { from, to, config },
```

Comment:
I think now that key and url are optional you can add a default on config here
```suggestion
  { from, to, config = {} },
```

Reply:
Done